library_generation/test/generate_library_unit_tests.sh

```diff
@@ -2,12 +2,13 @@
 
 set -xeo pipefail
 
+os_type=$1
 # Variables used to generate final result
 total_num=0
 succeed_num=0
 failed_num=0
 failed_tests=""
-# Unit tests against ./utilities.sh
+# Unit tests against utilities.sh and generate_library.sh
```

Comment:
I think it's better to differentiate the "true unit tests" for `utilities.sh` and "golden unit tests" for `generate_library.sh`. Maybe have a separate file for `generate_library_golden_unit_test`? Also mention that what are the scenarios we couldn't test just through the "true unit tests".

Reply:
I didn't use testing framework here so if we create another file for golden unit test for `generate_library.sh` I have to rewrite all test utility functions, like `assertEquals`.

How about I extract the test utilities to a separate file?

Reply:
> How about I extract the test utilities to a separate file?

Perfect idea!

Reply:
done.